packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -125,6 +108,13 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
 
   @override
   Future<void> init(InitParameters params) async {
+    // Throw if init() is called more than once
+    if (_initCalled.isCompleted) {
+      throw StateError(
+        'init() has already been called. Calling init() more than once results in undefined behavior.',
+      );
+    }
```

Comment:
This will prevent:
```dart
await plugin.init(...);
await plugin.init(...); // This will throw.
```
But will not prevent this:
```dart
plugin.init(...);
plugin.init(...); // This will NOT throw.
```

One way to solve it is to make `_initCalled` a `Completer<void>?` and set it here:

```suggestion
    if (_initCalled != null) {
      throw StateError(
        'init() has already been called. Calling init() more than once results in undefined behavior.',
      );
    }
    
    _initCalled = Completer<void>();
```